Update cartopy notebook

ajdawson opened the pull request:
This PR updates the cartopy notebook to display as slides using RISE. I've also added some extra content about the difference between `transform` and `projection`, and split a few things up so the talk flows more smoothly when presented as slides.

I've been using the latest Jupyter, so this change has also upgraded the notebook format.

duncanwp commented:
That looks good, thanks. The only thing I'd suggest is that the slide with the LATITUDE_FORMATTER is quite a lot of code to copy out. There's not much we can trim from it so we might just want to ask them not to work through it, or to copy it from the online version. 

The exercise looks good though. We should also point out at the end that Iris and CIS try to take care of transforms for you as much as possible.

ajdawson commented:
I guess they can modify their previous example to add the extra lines so they only have to type 2 lines extra (assuming they are using a notebook for this). I'll change the previous slide back to storing the gridliner in a variable to make this simpler.

ajdawson commented:
I've changed the gridliner example slightly so the first can be modified to become the second in student notebooks in a straightforward way. I also added an extra slide at the end briefly mentioning iris and cis.
